core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+        .create();
+
+    assertThat(catalog().listViews(ns1)).containsExactly(view1);
+    assertThat(catalog().listViews(ns2)).containsExactlyInAnyOrder(view2, view3);
+
+    assertThat(catalog().dropView(view2)).isTrue();
+    assertThat(catalog().listViews(ns1)).containsExactly(view1);
+    assertThat(catalog().listViews(ns2)).containsExactly(view3);
+
+    assertThat(catalog().dropView(view3)).isTrue();
+    assertThat(catalog().listViews(ns1)).containsExactly(view1);
+    assertThat(catalog().listViews(ns2)).isEmpty();
+
+    assertThat(catalog().dropView(view1)).isTrue();
+    assertThat(catalog().listViews(ns1)).isEmpty();
+    assertThat(catalog().listViews(ns2)).isEmpty();
+  }
+
+  @Test
+  public void listViewsAndTables() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    Namespace ns = Namespace.of("ns");
+
+    TableIdentifier tableIdentifier = TableIdentifier.of(ns, "table");
+    TableIdentifier viewIdentifier = TableIdentifier.of(ns, "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(ns);
+    }
+
+    assertThat(catalog().listViews(ns)).isEmpty();
+    assertThat(tableCatalog().listTables(ns)).isEmpty();
+
+    tableCatalog().buildTable(tableIdentifier, SCHEMA).create();
+    assertThat(catalog().listViews(ns)).isEmpty();
+    assertThat(tableCatalog().listTables(ns)).containsExactly(tableIdentifier);
+
+    catalog()
+        .buildView(viewIdentifier)
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(viewIdentifier.namespace())
+        .withQuery("spark", "select * from ns.tbl")
+        .create();
+
+    assertThat(catalog().listViews(ns)).containsExactly(viewIdentifier);
+    assertThat(tableCatalog().listTables(ns)).containsExactly(tableIdentifier);
+
+    assertThat(tableCatalog().dropTable(tableIdentifier)).isTrue();
+    assertThat(catalog().listViews(ns)).containsExactly(viewIdentifier);
+    assertThat(tableCatalog().listTables(ns)).isEmpty();
+
+    assertThat(catalog().dropView(viewIdentifier)).isTrue();
+    assertThat(catalog().listViews(ns)).isEmpty();
+    assertThat(tableCatalog().listTables(ns)).isEmpty();
+  }
+
+  @ParameterizedTest(name = ".createOrReplace() = {arguments}")
+  @ValueSource(booleans = {false, true})
+  public void createOrReplaceView(boolean useCreateOrReplace) {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    ViewBuilder viewBuilder =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .withProperty("prop1", "val1")
+            .withProperty("prop2", "val2");
```

Comment:
If you're testing the behavior of `createOrReplace`, then there is no need to test the full view metadata. That just makes the test longer. Instead, I'd recommend focusing on just the behavior you're validating here, which is the result of running `createOrReplace` or `replace`.

I would trim the assertions down to just the SQL that changes between the first create and the replace operation and the things that are expected to change, like a new schema and the view history.

Reply:
done